hw1/02-palindrome.js

```diff
@@ -1,3 +1,30 @@
-const elem = document.querySelector('input');
+const elem = document.querySelector("#num");
+elem.addEventListener("input", handleInput);
 
-elem.addEventListener('input', handleInput);
+function checkPalindrome(number) {
+  const num = String(number);
+  const backNum = num.split("").reverse().join("");
+  return num === backNum;
+}
+
+function checkNum(number) {
+  const answer = document.querySelector(".result");
+  answer.classList.remove("text-success", "text-danger");
+
+  if (isNaN(number) || number < 0) {
+    answer.textContent = "Please enter a positive number!";
+    answer.classList.add("text-danger");
+  } else {
+    const valid = checkPalindrome(number);
+
+    // Use ternary operator if ? then : else
+    answer.textContent = valid
+      ? "Yes. This is a palindrome!"
+      : "No. Try again.";
+    answer.classList.add(valid ? "text-success" : "text-danger");
```

Comment:
Careful with ternaries. They are often used in webdev because they do really cool things, but they can make reading code more complex than necessary. You're doing a good job here having it broken up this way though, I would just suggest moving the messages to a global `const` that can be referenced.

I believe in class we talked about how we "shouldn't" but everyone does and there's nothing horribly wrong with it. The hoisting doesn't necessarily take place I think, and it just allows a quick id of a var vs a whole string that - just like it has here - might push your linter into having to make a few new lines.

Reply:
I'll add as an additional note in general: consider writing more functions to be used in smaller snippets to do the small jobs you have happening. It can make debugging and testing much more manageable.

Reply:
Thanks for the feedback. As for ternaries, I'm careful when doing calculations, but since these are strings, I thought I would use its power. Okay, good to know because I'm still learning about hoisting haha. 

I will consider that in the following homework!